Doc/library/string.rst

```diff
@@ -232,6 +232,8 @@ are 0, 1, 2, ... in sequence, they can all be omitted (not just some)
 and the numbers 0, 1, 2, ... will be automatically inserted in that order.
 Because *arg_name* is not quote-delimited, it is not possible to specify arbitrary
 dictionary keys (e.g., the strings ``'10'`` or ``':-]'``) within a format string.
+The rules for parsing an item key is if it starts with a digit then it is treated as a
```

Comment:
I would prefer to change the wording on line 229-230 above instead: after the existing sentence, add "A key is treated as a number if it starts with a digit."

Reply:
Sure, updated.